File: samples/GluonSQLiteScala/src/main/scala/com.gluonhq.sqlite/GluonSQLite.scala
package com.gluonhq.sqlite

import com.gluonhq.charm.down.common.JavaFXPlatform
import com.gluonhq.charm.glisten.application.MobileApplication
import com.gluonhq.charm.glisten.visual.Swatch
import javafx.scene.Scene
import javafx.scene.image.Image
import javafx.stage.Stage
import com.gluonhq.sqlite._

class GluonSQLite extends MobileApplication {

    import MobileApplication._

    override def init(): Unit = {
        addViewFactory(HOME_VIEW, new BasicView(HOME_VIEW))
    }

    override def postInit(scene: Scene) {
        Swatch.RED.assignTo(scene)
        val stage = scene.getWindow.asInstanceOf[Stage]
        stage.getIcons.add(new Image(classOf[GluonSQLite].getResourceAsStream("/icon.png")))
        if (System.getProperty("os.arch").toUpperCase.contains("ARM") && !JavaFXPlatform.isIOS && !JavaFXPlatform.isAndroid) {
            stage.setFullScreen(true)
            stage.setFullScreenExitHint("")
        }
    }
}

File: samples/GluonSQLite/src/main/java/com/gluonhq/sqlite/GluonSQLite.java
package com.gluonhq.sqlite;

import com.gluonhq.charm.down.Platform;
import com.gluonhq.charm.glisten.application.MobileApplication;
import com.gluonhq.charm.glisten.visual.Swatch;
import javafx.scene.Scene;
import javafx.scene.image.Image;
import javafx.stage.Stage;

public class GluonSQLite extends MobileApplication {

    public static final String BASIC_VIEW = HOME_VIEW;

    @Override
    public void init() {
        addViewFactory(BASIC_VIEW, () -> new BasicView(BASIC_VIEW));
    }

    @Override
    public void postInit(Scene scene) {
        Swatch.BLUE.assignTo(scene);

        ((Stage) scene.getWindow()).getIcons().add(new Image(GluonSQLite.class.getResourceAsStream("/icon.png")));

        if (System.getProperty("os.arch").toUpperCase().contains("ARM") &&
                !Platform.isIOS()  && !Platform.isAndroid()) {
            ((Stage) scene.getWindow()).setFullScreen(true);
            ((Stage) scene.getWindow()).setFullScreenExitHint("");
        }
    }
}
